internal/cmd/controller/reconciler/cluster_controller.go

```diff
@@ -144,20 +150,21 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	if err != nil {
 		return ctrl.Result{}, r.updateErrorStatus(ctx, req.NamespacedName, cluster.Status, err)
 	}
-
-	deleted := map[types.UID]bool{}
-	for _, bundle := range cleanup {
-		for _, bd := range bundleDeployments.Items {
-			if bd.Labels[fleet.BundleLabel] == bundle.Name && bd.Labels[fleet.BundleNamespaceLabel] == bundle.Namespace {
-				logger.V(1).Info("cleaning up bundleDeployment not matching the cluster", "bundledeployment", bd)
-				err := r.Delete(ctx, &bd)
-				if err != nil {
-					logger.V(1).Error(err, "deleting bundleDeployment returned an error")
-				}
-				deleted[bd.GetUID()] = true
+	toDeleteBundles := indexByNamespacedName(cleanup)
+
+	// Delete BundleDeployments for Bundles being removed while getting a filtered items list
+	bundleDeployments.Items = slices.DeleteFunc(bundleDeployments.Items, func(bd fleet.BundleDeployment) bool {
+		bundleNamespace := bd.Labels[fleet.BundleNamespaceLabel]
+		bundleName := bd.Labels[fleet.BundleLabel]
+		if _, ok := toDeleteBundles[types.NamespacedName{Namespace: bundleNamespace, Name: bundleName}]; ok {
+			logger.V(1).Info("cleaning up bundleDeployment not matching the cluster", "bundledeployment", bd)
+			if err := r.Delete(ctx, &bd); err != nil {
+				logger.V(1).Info("deleting bundleDeployment returned an error", "error", err)
```

Comment:
@manno it seems like #3015 was not backported to 0.10 and this produced a conflict in this line.
Is it okay to keep the version that uses `Info` (like in 0.11) or should I change it back to use `Error`?:
```suggestion
				logger.V(1).Error(err, "deleting bundleDeployment returned an error")
```

Reply:
I've reconstructed the changes and used `logger.V(1).Error` to preserve the original behavior

Reply:
👍